### ISSUE TITLE
(VSCode) False positive of rule import-x/no-unresolved in monorepo with multiple packages open

### ISSUE THREAD

**Issue by auirarrazaval** [un-ts/eslint-plugin-import-x#370]
Hello! Im having the following issue when using VSCode with Eslint and Import-x. I'm opening this issue here because this is the only package where the issue arises, all other work properly. If this is not correct, please let me know

I have monorepo setup with a shared config file. there, I have "bundles" of plugins that my apps and packages can use. The bundle that sets up this extension is the following:

```ts
import { createTypeScriptImportResolver } from 'eslint-import-resolver-typescript';
import { importX } from 'eslint-plugin-import-x';
import noRelativeImportPaths from 'eslint-plugin-no-relative-import-paths';
import tseslint from 'typescript-eslint';

export default tseslint.config(
  importX.flatConfigs.recommended,
  importX.flatConfigs.typescript,
  {
    settings: {
      'import-x/resolver-next': [
        createTypeScriptImportResolver({
          project: [
            'tsconfig.json',
            'apps/*/tsconfig.json',
            'packages/*/tsconfig.json',
          ],
          alwaysTryTypes: true,
        }),
      ],
    },
  },
  {
    rules: { //... },
  },
);
```

This works well in the case I open just one app (lets say I open the workspace `apps/my-app-1` directly instead of the root directory of the monorepo. However, when I open the root directory, i have the following behavior:
 
- I open a file inside  `apps/my-app-1`, then the extension detects imports properly, without issues
- After opening the previous file, I open any file in `apps/my-app-2`, then the extension fails to find the local aliased imports:
![Image](https://github.com/user-attachments/assets/c84508cf-d55e-4ccb-9bcd-66bb1681ff54)

It seems to me like there's an issue with the path alias `@/` that is not being updated when I change apps inside the monorepo, but I could be totally wrong. 

I have created a small reproduction link, with instructions in the readme:
https://github.com/auirarrazaval/import-x-bug-no-unresolved-bug

**Comment by EDM115** [comment]
yup, started to see this recently too, and when running ESLint in the CLI it works perfectly fine

**Comment by EDM115** [comment]
@auirarrazaval I got really tired of issues between packages of the ESLint ecosystem, ever since the switch to the (very great) Flat config it seems like there's something that breaks every week, package authors release more updates than ever and sometimes it's hard to keep up.  
In the 16 weeks of my internship, i had to fix **4 times** issues between ESLint plugins, whether it being import, typescript or something else. Broken support in VS Code is a hassle too and getting it to work properly in a monorepo with shared configs is an absolute nightmare

I would really recommend to anyone trying to keep their sanity to switch to OxLint, they got most if not all rules you need (both from import-x and typescript-eslint), besides you should not rely on ESLint to catch TS issues, IDE integrations should get you covered and i'll advise for running a good `tsc --noEmit` alongside your lint commands to double-check

finally, if you really depend on plugins that don't have their rules ported to OxLint, keep ESLint but enable just those to remove the performance hit and all the goddamn inconsistencies between all those plugins.  
doing so saved us 2min40 in lint times (95%, now barely at 5s for 17 packages), dropped 118 dependencies (8%) and finally allows me to sleep without fearing of the next breakage of the lints right before I finish my internship

**Comment by JounQin** [comment]
I think you should report to https://github.com/microsoft/vscode-eslint instead.

**Comment by EDM115** [comment]
linked issue : https://github.com/microsoft/vscode-eslint/issues/1994

**Comment by JounQin** [comment]
@auirarrazaval Would you like to give the following a try?

```sh
# npm
npm i -D https://pkg.pr.new/eslint-plugin-import-x@388

# yarn
yarn add -D eslint-plugin-import-x@https://pkg.pr.new/eslint-plugin-import-x@388
```

See also https://github.com/un-ts/eslint-plugin-import-x/pull/388

**Comment by auirarrazaval** [comment]
@JounQin I have updated to the latest version (v4.16.0), which seems to include the change, but the error still persists. I'll try again later in the reproduction link I provided, but for now this didn't seem tot be enough for my codebase

**Comment by JounQin** [comment]
@auirarrazaval Sorry but I can't reproduce it in you repro, but there is indeed another regression described at #396, #397 and #398, and here is the fix #399.

Please help to test:

```sh
npm i https://pkg.pr.new/eslint-plugin-import-x@399
```

**Comment by konstantinkreft** [comment]
We have the same problem as @auirarrazaval and I just tested the #399 fix. It did not fix the issue in our repository.

**Comment by JounQin** [comment]
@SukkaW Can you reproduce it with the existing reproduction? I can't reproduce so I can't help debugging.

**Comment by auirarrazaval** [comment]
@JounQin Sorry! I've been a bit overwhelmed at work and hadn't had the time. I'll try to update the repo, i'll let you know

**Comment by NishargShah** [comment]
Hello @JounQin, this issue is still not fixed. I checked with the latest version and https://pkg.pr.new/eslint-plugin-import-x@399

Could you please take a look? If you’d like, I can provide a demo repo for reproduction.

**Comment by JounQin** [comment]
Sure if you can provide a reproduction that I can reproduce at the same time. I don't know what's the key difference yet.

**Comment by NishargShah** [comment]
Hello @JounQin, please check

## Repo

[https://github.com/NishargShah/import-x-monorepo-issue](https://github.com/NishargShah/import-x-monorepo-issue)

## Description

There seems to be a recurring issue with ESLint in this monorepo setup. The problem can be reproduced every time by following the steps below.

### Steps to Reproduce

**Initial setup (only once):**

1. Clone the repo.
2. Run `pnpm install`.

**Reproduction steps (repeat every time):**

1. Press `Cmd + Shift + P` → select **Developer: Reload Window**.
2. Open `/packages/package-1/src/index.ts`.
   
   - Go to line **5**.
   - Press **Enter** multiple times until ESLint activates.
   - Once empty lines are removed, ESLint works fine here.
3. Open `/packages/package-2/src/index.ts`.
   
   - Go to line **5**.
   - Press **Enter** multiple times.
   - You’ll see errors appear across all the imports.

## Screenshot

<img width="1726" height="993" alt="Image" src="https://github.com/user-attachments/assets/6dc91293-5148-4584-b8a4-c095ce9c9155" />

## Additional Note

1. `pnpm lint:fix` also works, so no eslint issues from CLI

<img width="1306" height="601" alt="Image" src="https://github.com/user-attachments/assets/2fc577e1-b05a-41c9-8a6f-5102a5987ace" />


I hope this will help you to find the issue

**Comment by NishargShah** [comment]
Any update @JounQin ?

**Comment by krish10122003** [comment]
Any update @JounQin?, its giving us the false warnings in vscode when switching from one package to another

**Comment by jianxingxuejian** [comment]
@NishargShah I cloned your repo and added a vscode config:
“eslint.useFlatConfig”: false
The issue disappeared.

By the way, I resolved it in my own project by changing it to:
“eslint.useFlatConfig”: true

**Comment by NishargShah** [comment]
Not working for me @jianxingxuejian, `eslint.useFlatConfig: false` will turn off the eslint flat config and none of the rules will work after setting it.

**Comment by NishargShah** [comment]
Hello @JounQin 

I have simplified the code and installed only required dependencies to produce the issue, please check

```
  "devDependencies": {
    "@eslint/compat": "^2.0.0",
    "@eslint/js": "^9.39.2",
    "eslint-import-resolver-typescript": "^4.4.4",
    "eslint-plugin-import-x": "^4.16.1",
    "typescript-eslint": "^8.52.0"
  }
```

**Comment by NishargShah** [comment]
@SukkaW Can you please help me with this issue, I am daily facing this issue and I am confused, where I need to raise the issue, in VSCode or here
